nebullvm/operations/base.py

```diff
@@ -0,0 +1,59 @@
+import abc
+import logging
+from typing import Dict, Any, Optional, Union
+
+from nebullvm.tools.base import Device
+from nebullvm.tools.utils import gpu_is_available
+
+logger = logging.getLogger("nebullvm_logger")
+
+
+def _check_device(device: Optional[str]) -> Device:
+    if device is None:
+        if gpu_is_available():
+            device = Device.GPU
+        else:
+            device = Device.CPU
+    else:
+        if device.lower() == "gpu":
+            if not gpu_is_available():
+                logger.warning(
+                    "Selected GPU device but no available GPU found on this "
+                    "platform. CPU will be used instead. Please make sure "
+                    "that the gpu is installed and can be used by your "
+                    "framework."
+                )
+                device = Device.CPU
+            else:
+                device = Device.GPU
+        else:
+            device = Device.CPU
+
+    return device
+
+
+class Operation(abc.ABC):
+    def __init__(self):
+        self._state = {}
+        self.device = Device.CPU
+        self.execute_count = 0
+        self.logger = logging.getLogger("nebullvm_logger")
```

Comment:
couldn't you just assign the global logger variable?

Reply:
Yes we can, but in this way every time an operation bust log something, it can log using the logger attribute without defining a global variable in each file